File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int maxWater = 0, minHeight = 0;
        int l = 0, r = height.length-1;
        while(l < r) {
            minHeight = Math.min(height[l], height[r]);
            int water = minHeight*(r-l);
            maxWater = Math.max(water, maxWater);
            if(minHeight == height[l]) l++;
            else r--;
        }
        return maxWater;
    }
}

File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        maxWater = 0
        minHeight = 0

        l = 0
        r = len(height)-1

        while l < r:
            minHeight = min(height[l], height[r])
            water = minHeight * (r-l)
            maxWater = max(maxWater, water)
            if minHeight == height[l]:
                l+=1
            else:
                r-=1

        return maxWater
        